src/jit/liveness.cpp

```diff
@@ -2185,11 +2185,10 @@ bool Compiler::fgRemoveDeadStore(GenTree**        pTree,
                     {
                         rhsNode = asgNode->AsBlk()->Data();
                     }
-                    // TODO-1stClassStructs: There should be an else clause here to handle
-                    // the non-block forms of store ops (GT_STORE_LCL_VAR, etc.) for which
-                    // rhsNode is op1. (This isn't really a 1stClassStructs item, but the
-                    // above was added to catch what used to be dead block ops, and that
-                    // made this omission apparent.)
+                    else
+                    {
+                        rhsNode = asgNode->gtGetOp2();
```

Comment:
Could you please add a comment about what do we expect here?
The old comment said that `rhsNode is op1`, but here you are using `gtGetOp2()`, was the comment wrong?

Also while you are here could you please look at the code that starts this block:
https://github.com/dotnet/coreclr/blob/2520798548b0c414f513aaaf708399f8ef5a4f6c/src/jit/liveness.cpp#L2164-L2172

It looks wrong that we check `if (asgNode == nullptr) ` twice, should it be:
```
if (asgNode == nullptr)
{
  if (addrNode == nullptr)
  {
    asgNode = nextNode;
  }
  if (asgNode == nullptr)
  {
```
or a separate issue?

Reply:
That is a separate issue. I'll open a new issue to fix that, as well as clarifying the code that I added. I don't really want to re-push and add more overload to the already-overloaded CI system.

Reply:
Nevermind. The PR testing was stalled anyway, so I pushed with some clarifying comments (and deleting the extra check for asgNode being non-null).

Reply:
Thank you.